tests/NLog.UnitTests/Targets/FileTargetTests.cs

```diff
@@ -120,25 +120,31 @@ public void SimpleFileTest(bool concurrentWrites, bool keepFileOpen, bool networ
             }
         }
 
-#if !MONO
         [Theory]
-#else
-        [Theory(Skip="Not supported on MONO on Travis, because of FileSystemWatcher not working")]
-#endif
         [MemberData(nameof(SimpleFileTest_TestParameters))]
         public void SimpleFileDeleteTest(bool concurrentWrites, bool keepFileOpen, bool networkWrites, bool forceManaged, bool forceMutexConcurrentWrites, bool optimizeBufferReuse)
         {
+            bool isSimpleKeepFileOpen = keepFileOpen && !networkWrites && !concurrentWrites && IsTravis();
+            if (IsTravis())
```

Comment:
Well the logic is that `SimpleFileDeleteTest` works on Travis when `!concurrentWrites`. So the conditions are very correct. Maybe hard to read on mobile.

Reply:
Maybe I'm missing something, but why is in the isSimpleKeepFileOpen a isTravis-check while the whole unit test is skipped on Travis ;)

Reply:
Please join the nested ifs, thanks :)

Reply:
Not sure I understand. `isSimpleKeepFileOpen` is used outside the `if (IsTravis())`

Reply:
And I love nested ifs. Never understood why they are frowned upon.

Reply:
I think unneeded nesting is a code smell.

Reply:
> Merging collapsible if statements increases the code's readability.

https://rules.sonarsource.com/csharp/RSPEC-1066

Reply:
Well I like different logical things on different lines. `IsTravis()` is the checking the platform. The nested-if checks the test-scenario. For me two different logical checks. But no need to win this, since just a unit-test. One if-statement it is.

Reply:
It's not about winning, but in another PR I was busy removing all (unneeded) nested ifs ;)

(It will give a sonar warning)

Reply:
Well I don't agree with all the Sonar issues. I understand 4 level nested if-statements is a code-smell. But 2 levels is often more readable instead of one big chunk.

Reply:
I think they are promoting readability with conditions by variables/methods instead of nesting.